From 389d260ca41a277de49ca0154966d549945d8635 Mon Sep 17 00:00:00 2001
From: Kyle McMartin <kmcmarti@redhat.com>
Date: Sun, 16 Jun 2013 12:48:46 -0400
Subject: [PATCH] debugedit: handle aarch64 debug_info relocations
 (RhBug:974860)

AArch64 generates a relocation which must be handled similar to other
architectures. Adding this patch allows debugedit to run against the
kernel debuginfo.

Signed-off-by: Panu Matilainen <pmatilai@redhat.com>
---
 debugedit.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/debugedit.c b/debugedit.c
index a658d5b..434bab0 100644
--- a/debugedit.c
+++ b/debugedit.c
@@ -1158,6 +1158,10 @@ edit_dwarf2 (DSO *dso)
 		  if (rtype != R_ALPHA_REFLONG)
 		    goto fail;
 		  break;
+		case EM_AARCH64:
+		  if (rtype != R_AARCH64_ABS32)
+		    goto fail;
+		  break;
 		default:
 		fail:
 		  error (1, 0, "%s: Unhandled relocation %d in .debug_info section",
